### ISSUE TITLE
Windows kubelet continually recreate secret and configure in pod before kubernetes 1.15

### ISSUE THREAD

**Issue by aiwantaozi** [rancher/rancher#22145]
Rancher server: master
Step to reproduce: 
1. add a cluster with enabling windows(Host gateway), cluster version 1.14.4
2. use view in API to edit kubelet params, add extra-arg to configure the log level to 6
```
 "kubelet": {
            "extraArgs": {
                "v": "6"
            },
            "failSwapOn": false,
            "type": "/v3/schemas/kubeletService"
        },
```
3. create a secret named viky with key viky, value "xxx"
4. create a deployment with image ```golang:1.12-windowsservercore``` mount that secret and schedule to windows node 
![image](https://user-images.githubusercontent.com/5936271/62942274-88779700-be0a-11e9-8c83-223da1001d44.png)
5. check kubelet logs in windows node `` docker logs kubelet_container_id | select-string viky |Select-String atomic_writer```
6. could see even we don't update the secret, kubelet will treat that the secret is updated, could see log `performed write of new data to ts data directory`

more info:
logs for windows kubelet:
```
I0813 11:36:19.980191   15192 atomic_writer.go:332] c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1: current paths:
[c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1\..2019_08_13_11_35_18.220380704\viky]
I0813 11:36:19.980191   15192 atomic_writer.go:344] c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1: new paths:       [viky]
I0813 11:36:19.980191   15192 atomic_writer.go:347] c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1: paths to remove:
map[c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1\..2019_08_13_11_35_18.220380704\viky:{}]
I0813 11:36:19.980191   15192 atomic_writer.go:160] pod test/viky-64567b48dc-wdqnh volume vol1: write required for target directory
c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1
I0813 11:36:19.981190   15192 atomic_writer.go:160] pod test/viky-64567b48dc-wdqnh volume default-token-hlstr: write required for target directory
c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\default-token-hlstr
I0813 11:36:19.982190   15192 atomic_writer.go:177] pod test/viky-64567b48dc-wdqnh volume vol1: performed write of new data to ts data directory:
c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1\..2019_08_13_11_36_19.998145594
I0813 11:36:19.984191   15192 atomic_writer.go:177] pod test/viky-64567b48dc-wdqnh volume default-token-hlstr: performed write of new data to ts data directory:
c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\default-token-hlstr\..2019_08_13_11_36_19.224743505
I0813 11:37:30.953188   15192 atomic_writer.go:332] c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1: current paths:
[c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1\..2019_08_13_11_36_19.998145594\viky]
I0813 11:37:30.953188   15192 atomic_writer.go:344] c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1: new paths:       [viky]
I0813 11:37:30.953188   15192 atomic_writer.go:347] c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1: paths to remove:
map[c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1\..2019_08_13_11_36_19.998145594\viky:{}]
I0813 11:37:30.953188   15192 atomic_writer.go:160] pod test/viky-64567b48dc-wdqnh volume vol1: write required for target directory
c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1
I0813 11:37:30.954190   15192 atomic_writer.go:160] pod test/viky-64567b48dc-wdqnh volume default-token-hlstr: write required for target directory
c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\default-token-hlstr
I0813 11:37:30.956189   15192 atomic_writer.go:177] pod test/viky-64567b48dc-wdqnh volume vol1: performed write of new data to ts data directory:
c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1\..2019_08_13_11_37_30.989197686
I0813 11:37:30.959154   15192 atomic_writer.go:177] pod test/viky-64567b48dc-wdqnh volume default-token-hlstr: performed write of new data to ts data directory:
c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\default-token-hlstr\..2019_08_13_11_37_30.233828957
```

logs for linux kubelet:
```
{"log":"I0813 10:40:45.304113    4521 atomic_writer.go:157] pod test/viky2-6564554cf9-swtqg volume default-token-hlstr: no update required for target directory /var/lib/kubelet/pods/cbf5bcfb-bdb6-11e9-a369-0a5d188d99c0/volumes/kubernetes.io~secret/default-token-hlstr\n","stream":"stderr","time":"2019-08-13T10:40:45.304186682Z"}
{"log":"I0813 10:40:45.304331    4521 atomic_writer.go:332] /var/lib/kubelet/pods/cbf5bcfb-bdb6-11e9-a369-0a5d188d99c0/volumes/kubernetes.io~secret/viky2: current paths:   [viky2]\n","stream":"stderr","time":"2019-08-13T10:40:45.304400999Z"}
{"log":"I0813 10:40:45.304363    4521 atomic_writer.go:344] /var/lib/kubelet/pods/cbf5bcfb-bdb6-11e9-a369-0a5d188d99c0/volumes/kubernetes.io~secret/viky2: new paths:       [viky2]\n","stream":"stderr","time":"2019-08-13T10:40:45.304408867Z"}
{"log":"I0813 10:40:45.304371    4521 atomic_writer.go:347] /var/lib/kubelet/pods/cbf5bcfb-bdb6-11e9-a369-0a5d188d99c0/volumes/kubernetes.io~secret/viky2: paths to remove: map[]\n","stream":"stderr","time":"2019-08-13T10:40:45.304448484Z"}
{"log":"I0813 10:40:45.304401    4521 atomic_writer.go:157] pod test/viky2-6564554cf9-swtqg volume viky2: no update required for target directory /var/lib/kubelet/pods/cbf5bcfb-bdb6-11e9-a369-0a5d188d99c0/volumes/kubernetes.io~secret/viky2\n","stream":"stderr","time":"2019-08-13T10:40:45.304458206Z"}
{"log":"I0813 10:40:46.306965    4521 atomic_writer.go:332] /var/lib/kubelet/pods/cbf5bcfb-bdb6-11e9-a369-0a5d188d99c0/volumes/kubernetes.io~secret/viky2: current paths:   [viky2]\n","stream":"stderr","time":"2019-08-13T10:40:46.307065642Z"}
{"log":"I0813 10:40:46.306996    4521 atomic_writer.go:344] /var/lib/kubelet/pods/cbf5bcfb-bdb6-11e9-a369-0a5d188d99c0/volumes/kubernetes.io~secret/viky2: new paths:       [viky2]\n","stream":"stderr","time":"2019-08-13T10:40:46.307079338Z"}
{"log":"I0813 10:40:46.307003    4521 atomic_writer.go:347] /var/lib/kubelet/pods/cbf5bcfb-bdb6-11e9-a369-0a5d188d99c0/volumes/kubernetes.io~secret/viky2: paths to remove: map[]\n","stream":"stderr","time":"2019-08-13T10:40:46.30708486Z"}
{"log":"I0813 10:40:46.307022    4521 atomic_writer.go:157] pod test/viky2-6564554cf9-swtqg volume default-token-hlstr: no update required for target directory /var/lib/kubelet/pods/cbf5bcfb-bdb6-11e9-a369-0a5d188d99c0/volumes/kubernetes.io~secret/default-token-hlstr\n","stream":"stderr","time":"2019-08-13T10:40:46.307089857Z"}
{"log":"I0813 10:40:46.307036    4521 atomic_writer.go:157] pod test/viky2-6564554cf9-swtqg volume viky2: no update required for target directory /var/lib/kubelet/pods/cbf5bcfb-bdb6-11e9-a369-0a5d188d99c0/volumes/kubernetes.io~secret/viky2\n","stream":"stderr","time":"2019-08-13T10:40:46.307094683Z"}
```

Could see that `current paths` in linux is just viky, but in windows it include the prefix `c:\var\lib\kubelet\pods\d4fd876f-bdb3-11e9-a369-0a5d188d99c0\volumes\kubernetes.io~secret\vol1\..2019_08_13_11_36_19.998145594\viky` which makes this function always have result to be deleted, actually it shouldn't be deleted. Seems like the trimPrefix not work here https://github.com/kubernetes/kubernetes/blob/v1.14.4/pkg/volume/util/atomic_writer.go#L317, 

Because the secret and configmap continually be recreated, which affect the logging support in Windows. Logging watch secret file to check whether the configure was updated, then reload to use the latest configure. 

After this commit https://github.com/kubernetes/kubernetes/commit/78d109e20157a24012059bcf154abf4320bd513f (After release 1.15.0). This secret and configmap isn't continually recreate in windows. Need to pay attention to that.

**Comment by loganhz** [comment]
It should be fixed in kubernetes side

**Comment by loganhz** [comment]
This only happens in k8s 1.14. It has been fixed in 1.15
